protocols/gossipsub/src/handler.rs

```diff
@@ -383,6 +404,13 @@ impl EnabledHandler {
             }
         }
 
+        // Drop the next message in queue if it's stale.
+        if let Poll::Ready(Some(rpc)) = self.send_queue.poll_stale(cx) {
+            return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                HandlerEvent::MessageDropped(rpc),
+            ));
+        }
```

Comment:
Drain all stale messages in the queue?
```suggestion
        // Drop stale messages in the queue.
        while let Poll::Ready(Some(rpc)) = self.send_queue.poll_stale(cx) {
            return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
                HandlerEvent::MessageDropped(rpc),
            ));
        }
```

Reply:
I think it's the same no? Since whenever the condition matches the function will return, that's why I prefer `if`, `while` gives a wrong impression

Reply:
Makes sense :+1: